src/_pytest/warnings.py

```diff
@@ -123,7 +123,7 @@ def warning_record_to_str(warning_message):
     if unicode_warning:
         warnings.warn(
             "Warning is using unicode non convertible to ascii, "
-            "converting to a safe representation:\n  %s" % msg,
+            "converting to a safe representation:\n  %s" % compat.safe_str(msg),
```

Comment:
I'd probably use `{!r}` :man_shrugging:

Reply:
hah -- I suggested this too.  at least I'm consistent: https://github.com/pytest-dev/pytest/issues/3691#issuecomment-405766850